src/host/mapreduce.jl

```diff
@@ -1,9 +1,14 @@
 # map-reduce
 
+const AbstractArrayOrBroadcasted = Union{AbstractArray,Broadcast.Broadcasted}
+
 # GPUArrays' mapreduce methods build on `Base.mapreducedim!`, but with an additional
 # argument `init` value to avoid eager initialization of `R` (if set to something).
-mapreducedim!(f, op, R::AbstractGPUArray, As::AbstractArray...; init=nothing) = error("Not implemented") # COV_EXCL_LINE
+mapreducedim!(f, op, R::AbstractGPUArray, A::AbstractArrayOrBroadcasted;
+              init=nothing) = error("Not implemented") # COV_EXCL_LINE
+# resolve ambiguities
```

Comment:
@tkf I guess you didn't run into such ambiguities with https://github.com/JuliaLang/julia/pull/31020?

Reply:
e.g., for the `mapreduce` case below:

```
ERROR: MethodError: mapreduce(::typeof(+), ::typeof(+), ::JLArray{Float64,1}, ::JLArray{Float64,2}) is ambiguous. Candidates:
  mapreduce(f, op, A::AbstractArray...; kw...) in Base at reducedim.jl:308
  mapreduce(f, op, A::Union{Base.Broadcast.Broadcasted, AbstractArray}, As::Union{Base.Broadcast.Broadcasted, AbstractArray}...; dims, init) in GPUArrays at /home/tim/Julia/pkg/GPUArrays/src/host/mapreduce.jl:27
Possible fix, define
  mapreduce(::Any, ::Any, ::AbstractArray, ::Vararg{AbstractArray,N} where N)
```

Reply:
I don't remember seeing anything similar to this but it's a long time ago that I wrote that patch.